src/Tribe/Utils/Element_Classes.php

```diff
@@ -0,0 +1,249 @@
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` class="class1 class2" `
+	 */
+	public function __invoke() {
+		$this->arguments = func_get_args();
+		return $this->get_attribute();
+	}
+
+
+	/**
+	 * When cast to string an instance will return the full HTML class attribute.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` class="class1 class2" `
+	 */
+	public function __toString() {
+		return $this->get_attribute();
+	}
+
+	/**
+	 * Gets the full HTML class attribute for this instance of Element Classes.
+	 * It will contain a space on each end of the attribute.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` class="class1 class2" `
+	 */
+	public function get_attribute() {
+		$classes = $this->get_classes_as_string();
+
+		// Bail with empty string when no classes are present
+		if ( ! $classes ) {
+			return '';
+		}
+
+		return " class=\"{$classes}\" ";
+	}
+
+	/**
+	 * Gets a space separated string of all classes to be printed.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string
+	 */
+	public function get_classes_as_string() {
+		return implode( ' ', $this->get_classes() );
+	}
+
+	/**
+	 * Get the array of classes to be printed.
+	 *
+	 * @since  TBD
+	 *
+	 * @return array
+	 */
+	public function get_classes() {
+		$this->results = [];
+		$classes       = [];
+
+		$this->parse_array( $this->arguments );
+
+		foreach ( $this->results as $key => $val ) {
+			if ( ! $val ) {
+				continue;
+			}
+
+			$classes[] = $key;
+		}
+
+		$classes = array_map( 'sanitize_html_class', $classes );
+		$classes = array_filter( array_unique( $classes ) );
+
+		return $classes;
+	}
+
+
+	/**
+	 * Get the array of the classes, using [ class_name => bool ] as the format.
+	 *
+	 * @since  TBD
+	 *
+	 * @return array [ class_name => bool ]
+	 */
+	public function get_conditions() {
+		$this->results = [];
+		$this->parse_array( $this->arguments );
+
+		return $this->results;
+	}
+
+	/**
+	 * Parse arguments or argument for this instance, and store values on results.
+	 *
+	 * @since  TBD
+	 *
+	 * @param  mixed  $arguments      Any possible set of arguments that this class supports.
+	 * @param  boolean $default_value What is the default value for a given class.
+	 *
+	 * @return void
+	 */
+	protected function parse( $arguments, $default_value = true ) {
+		if ( ! $arguments ) {
+			return;
+		}
+
+		if ( is_string( $arguments ) ) {
+			// 'foo bar'
+			$this->parse_string( $arguments );
+		} elseif ( $arguments instanceof \Closure || is_callable( $arguments ) ) {
+			// function() {}
+			$this->parse_callable( $arguments );
+		} elseif ( is_array( $arguments ) ) {
+			// ['foo', 'bar', ...] || ['foo' => TRUE, 'bar' => FALSE, ...]
+			$this->parse_array( $arguments );
+		} elseif ( is_object( $arguments ) ) {
+			// stdClass
+			$this->parse_object( $arguments );
+		} elseif ( is_int( $arguments ) ) {
+			// '130'
+			$this->results[ $arguments ] = $default_value;
+		} elseif ( is_float( $arguments ) ) {
```

Comment:
Classes cannot start with a digit so I'm not sure this is valid?
ie: Converting, say, `2.5` to a string does not make it a valid CSS class.

I'd suggest that the logic throw out any classes that start with `a digit, two hyphens, or a hyphen followed by a digit` as those are all invalid? Or maybe a separate`validate_class` function.

Also worth noting that WP `sanitize_html_class` (above) does not validate classes - it only strips out illegal characters (and it needs updating)

Reply:
Since we are relying on the WP method, I would not care about that particular case. Unless we ported their code internally and applied rules. What do you think? Should we implement our own?

Reply:
I'd still remove this case - unless there is a compelling reason to allow floats as classes? (I expect not since they are invalid) - or tell it to bail on a float, etc.

I'd start the `if-else` chain with checks for:
```
if ( substr( $string, 0, 2 ) === '--') { 
// starts with two hyphens - bail
} elseif( is_numeric( substr( $string, 0, 1  )) ) {
// starts with a number - bail
} elseif( substr( $string, 0, 1 ) === '-` && is_numeric( substr( $string, 1, 1 ) ) ) {
// starts with a hyphen + a number  - bail
}
```
Since once it's out there we have no control over what gets passed to it via hooks, etc, it seems prudent to put those sanity checks in?